File: Cap1CSharp/ProcessosPosLINQ.cs
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;

namespace Cap1CSharp 
{
    public class ProcessosPosLINQ 
    {
        public IEnumerable<Process> ImprimeProcessos()
        {
            var processos = from p in Process.GetProcesses()
                            where p.ProcessName.StartsWith("w")
                            select p;
            return processos;
        }
    }
}

File: Cap1VB/ProcessosPosLINQ.vb
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Linq

Public Class ProcessosPosLINQ
  Public Function ImprimeProcessos() As IEnumerable(Of Process)
    Dim processos = From p In Process.GetProcesses() _
      Where p.ProcessName.StartsWith("w") _
      Select p
    Return processos
  End Function
End Class